src/_pytest/nodes.py

```diff
@@ -666,6 +666,19 @@ def isinitpath(self, path: Union[str, "os.PathLike[str]"]) -> bool:
         return self.session.isinitpath(path)
 
 
+class Directory(FSCollector):
+    """Base class for collecting files from a directory.
+
+    A basic directory collector does the following: goes over the files and
+    sub-directories in the directory and creates collectors for them by calling
+    the hooks :hook:`pytest_collect_directory` and :hook:`pytest_collect_file`,
+    after checking that they are not ignored using
+    :hook:`pytest_ignore_collect`.
```

Comment:
Ping here @bluetech --  in case my comment was not clear, I think we should highlight why we have `Dir` and `Directory`, given the names are so similar.

Reply:
Changes the docstrings a bit, I tried to leave enough breadcrumbs for the reader of any of the 3 classes `Directory`, `Dir` and `Package` to figure out how they relate.